tests/pipeline/test_engine.py

```diff
@@ -1242,6 +1249,177 @@ def test_dollar_volume(self):
+    def test_correlation_factors(self, returns_length, correlation_length):
+        """
+        Tests for the built-in factors `RollingPearsonOfReturns` and
+        `RollingSpearmanOfReturns`.
+        """
+        my_asset_column = 0
+        start_date_index = 6
+        end_date_index = 10
+
+        assets = self.asset_finder.retrieve_all(self.sids)
+        my_asset = assets[my_asset_column]
+        my_asset_filter = (AssetID() != (my_asset_column + 1))
+        num_assets = len(self.sids)
+        num_days = end_date_index - start_date_index + 1
+
+        # Our correlation factors require that their baseline asset is not
+        # filtered out, so make sure that masking out our baseline asset does
+        # not take effect. That is, a filter which filters out only our
+        # baseline asset should produce the same result as if no mask was
+        # passed at all.
+        for mask in (NotSpecified, my_asset_filter):
+            for CorrelationFactor, correlation_function in (
+                    (RollingPearsonOfReturns, pearsonr),
+                    (RollingSpearmanOfReturns, spearmanr)):
+                correlation = CorrelationFactor(
+                    baseline_asset=my_asset,
+                    returns_length=returns_length,
+                    correlation_length=correlation_length,
+                    mask=mask,
+                )
+
+                results = self.engine.run_pipeline(
+                    Pipeline(columns={'correlation': correlation}),
+                    self.dates[start_date_index],
+                    self.dates[end_date_index],
+                )
+                correlation_results = results['correlation'].unstack()
+
+                # Run a separate pipeline that calculates returns starting
+                # (correlation_length - 1) days prior to our start date. This
+                # is because we need (correlation_length - 1) extra days of
+                # returns to compute our expected correlations.
+                returns = Returns(window_length=returns_length)
+                results = self.engine.run_pipeline(
+                    Pipeline(columns={'returns': returns}),
+                    self.dates[start_date_index - (correlation_length - 1)],
+                    self.dates[end_date_index],
+                )
+                returns_results = results['returns'].unstack()
+
+                # On each day, calculate the expected correlation coefficient
+                # between the asset we are interested in and each other asset.
+                # Each correlation is calculated over `correlation_length`
+                # days.
+                expected_correlation_results = full_like(
+                    correlation_results, nan,
+                )
+                for day in range(num_days):
+                    my_asset_returns = returns_results.ix[
+                        day:day+correlation_length, my_asset_column,
```

Comment:
flake8 is going to yell at you about not having whitespace around the `+` here I think.